rasa/core/brokers/kafka.py

```diff
@@ -212,6 +212,9 @@ def _convert_intent_id_to_string(self, event: Dict[Text, Any]) -> Dict[Text, Any
             event["parse_data"]["intent"]["id"] = str(
                 event["parse_data"]["intent"]["id"]
             )
+            for idx, parse_data in enumerate(event["parse_data"]["intent_ranking"]):
+                parse_data["id"] = str(parse_data["id"])
+                event["parse_data"]["intent_ranking"][idx] = parse_data
```

Comment:
I don't think you need this line (or the `enumerate`) as you are modifying the dict in the line above.

Reply:
Ah so `intent_ranking` is a separate field from `intent` and contains the full breakdown of intents + ID's

Reply:
Sorry missed this before was merged.
I mean that `parse_data` is a direct reference to `event["parse_data"]["intent_ranking"][idx]` so when you update `parse_data["id"]` you are also also updating `event["parse_data"]["intent_ranking"][idx]["id"]` so no need to re-assign  `parse_data` to `event["parse_data"]["intent_ranking"][idx]`. And thus you no longer need to know `idx`.

Reply:
Ah makes sense, I'll create another PR to get this updated